moabb/paradigms/motor_imagery.py

```diff
@@ -24,54 +27,68 @@ class BaseMotorImagery(BaseParadigm):
         Low cut-off frequency in Hz. If None the data are only low-passed.
     fmax : float | None, (default 35)
         High cut-off frequency in Hz. If None the data are only high-passed.
+    interval: list | None, (default None)
+        Interval to cut trial. If None, defaults to the dataset-defined interval
 
     """
 
-    def __init__(self, fmin=7, fmax=35, channels=None, **kwargs):
+    def __init__(self, fmin=7, fmax=35, channels=None, interval=None, **kwargs):
         """init"""
         super().__init__(**kwargs)
         self.fmin = fmin
         self.fmax = fmax
         self.channels = channels
+        self.interval = interval
+        if self.interval is not None:
+            assert self.interval[0] >= 0
 
     def verify(self, dataset):
         '''
         Method that verifies dataset is correct for given parameters
         '''
         assert dataset.paradigm == 'imagery'
+        if self.interval is not None:
+            assert dataset.interval[1] > self.interval[1]
 
-    def _epochs(self, raws, event_dict, time):
-        '''Take list of raws and returns a list of epoch objects. Implements
+    def raw_to_trials(self, raw, events, picks, event_dict, time):
+        '''
+        Given a single raw file turn it into a features matrix and labels
+        '''
+        raw_f = raw.filter(self.fmin, self.fmax, method='iir', picks=picks,
```

Comment:
`raw.copy().filter()`, to be safe. filter modify the data inplace

Reply:
thanks for the catch!